pkg/reports/repo.go

```diff
@@ -0,0 +1,80 @@
+package reports
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+
+	"github.com/moov-io/customers/pkg/client"
+)
+
+func NewRepository(logger log.Logger, db *sql.DB) *sqlRepository {
+	return &sqlRepository{logger: logger, db: db}
+}
+
+type Repository interface {
+	FindByAccountIDs(accountIDs []string) ([]*CustomerAccount, error)
+}
+
+type sqlRepository struct {
+	db     *sql.DB
+	logger log.Logger
+}
+
+func (r *sqlRepository) FindByAccountIDs(accountIDs []string) ([]*CustomerAccount, error) {
+	query := fmt.Sprintf(
+		`select
```

Comment:
Why would we duplicate the logic for reading a `Customer`? That's pretty involved logic to duplicate.

Reply:
We aren't reading out address, phone or metadata either.

Reply:
Agreed, I was debating on whether to implement this. Based on the issue description, it seemed like we wanted to minimize the # of db queries we were making for fetching `n` # of customers/accounts. So it wasn't too much code to duplicate.

I could rip it out and call `getCustomer` * `n`. I also looked into using `searchCustomers` but underneath it still calls `getCustomer` `n` times.

And I assumed address, phone or metadata of the customer wasn't relevant to this endpoint.

Reply:
I would always default to assuming we need to return the entire model. Otherwise the UI and folks could get confused.

It sounds like we need to reinvent the repository method for getting multiple customers then? I don't like this hack when we could have one method used in `GET /customers/....` routes and this reporting endpoint.

Reply:
Good point.

I'll rework this, I'll use `searchCustomers` for now but will optimize it in another PR.

Related issue: #216